generate-github-io

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+cdir=`pwd`
+branch=`git rev-parse --abbrev-ref HEAD`
+clone=~/trydent-temp-dir/TrydentEngine
+archive=~/trydent-temp-dir/temp-trydent-javadoc.tar
+
+mkdir ~/trydent-temp-dir
+
+# This is here so that if something goes wrong, we don't erase
+# anything important.
+mkdir safety-dir
+cd safety-dir
+
+git clone https://github.com/AaronFriesen/Trydent.git $clone
+cd $clone
+git checkout $branch
+gradle clean
+gradle javadoc
+cd build/docs/javadoc
+tar -cvf $archive *
+cd $clone
+git checkout gh-pages
+rm -rf "$clone/*"
+git ls-files --deleted | xargs git rm
+tar -xvf $archive 
+tar -tf $archive | xargs git add
+rm "$archive"
+git commit -am "Updating javadocs for github.io."
+git push origin gh-pages
```

Comment:
This pushes every time, regardless of whether we're on master or a temporary working branch. I can imagine situations where this might be run (either accidentally or without thinking about it) and update the github.io with incorrect or even broken javadocs. How do we want to handle this?